docker-compose.yml

```
@@ -3,17 +3,12 @@ version: '3.8'
 services:
   web:
     build: .
-    command: >
-+      sh -c "python manage.py collectstatic --no-input &&
-+      gunicorn mysite.wsgi:application --bind 0.0.0.0:80"
     ports:
       - 80:80
     env_file:
       - ./.env
     depends_on:
       - db
```

```suggestion
      - db
  volumes:
      - ./:/usr/src/app
```

Pero lo mas fundamental que te falta aca es mapear el directorio local. Esto permite que el container se entere de que hay cambio en el codigo mientras estas "codeando" en tu ambiente local.

Para production no tiene mucho sentido, porque no vas a codear en el server, entonces el container puede correr la version de codigo que hay en la imagen... estoy es stateless, no hay nada en la imagen que pueda cambiar y sea distinto el comportamiento
Pero para development queres cambiar, queres hacer reload de cambios y eso se obtiene montando el volumen

Tambien podes dejar asi para production, no afecta

Te ahorras de hacer `docker compose build web` todo el tiempo